core/contracts/common/implementation/MultiRole.sol

```diff
@@ -35,6 +35,7 @@ library Shared {
     }
 
     function addMember(RoleMembership storage roleMembership, address memberToAdd) internal {
+        require(memberToAdd != address(0x0), "Cannot set add shared role to 0x0");
```

Comment:
You know what's really frustrating? Coverage/non optimized compilation breaks because of this line, specifically because of the require string. If you remove it, the non optimized compilation works as expected (at least from my testing). Still looking into it -- definitely going to try to file a bug against solidity.

Reply:
Yeah, pretty confident that this is a solidity bug -- I reproduced a minimal example and filed it here: https://github.com/ethereum/solidity/issues/8656. For now, I think we should just put the require message in a comment along with a link to the bug.

Reply:
Thanks for looking into this! it's crazy how it breaks it like this and that removing the require message fixes it. how frustrating. 

After playing around a bit on my local setup it seems that this has something to do specifically with having a require message on an internal function. As these are internal I think it's fine to remove the require message for now.